### ISSUE TITLE
Is it needed to tar the complete target folder?

### ISSUE THREAD

**Issue by abrauner** [heroku/heroku-maven-plugin#10]
We got errors about too many open files when tar-ing the target folder. As there is already a way to include additional files it makes sense from my point of view to include only a single jar or war file and those additional includes. Test-Resources from the build should not be included into the slug, should they?
I would create a pull request where only a configurable-file is used as jar or war include if this is ok.

**Comment by jkutner** [comment]
Yes, I like your suggestion on this, but I'm afraid it will cause a surprise break for uses who are currently including things outside of the `target` directory. Maybe we only exclude target from the list if the `<include>` is a path to file(s) in the target directory?

**Comment by abrauner** [comment]
I'd suggest to make something like a 
```xml
<includeJarFileOnly>my-jar.jar</includeJarFileOnly>
```
where we can say if this is set we only copy that jar to the heroku/app folder.

**Comment by jkutner** [comment]
That's awfully specific. I'd prefer a more general approach that could apply to other cases. What about an `<includesOnly>` option that overrides everything. It might look like:

```xml
<includesOnly>
  <include>target/my-jarfile.jar</include>
</includesOnly>
```

Would that work for you?

**Comment by philwebb** [comment]
+1 for this issue. It would be very helpful when deploying a Spring Boot application which already contains everything need to run inside the main JAR.

**Comment by philwebb** [comment]
I've had a go at adding a `includeTarget` toggle property. See #13.

**Comment by jkutner** [comment]
Fixed by #13. This will be available in 0.3.4, which is already released and should be in repo1 in a few hours (or when ever it syncs).
